src/transaction/estimate_gas.rs

```diff
@@ -25,23 +56,21 @@ impl<T: Transport> EstimateGasFuture<T> {
         let eth = builder.web3.eth();
 
         let from = builder.from.map(|account| account.address());
-        let to = builder.to.unwrap_or_else(Address::zero);
-        let request = CallRequest {
+        let gas_price = builder.gas_price.and_then(|gas_price| gas_price.value());
+        let request = EstimateGasRequest {
             from,
-            to,
+            to: builder.to,
             gas: None,
-            gas_price: None,
+            gas_price,
             value: builder.value,
             data: builder.data,
         };
 
         EstimateGasFuture::from_request(eth, request)
     }
 
-    /// Create an estimate gas future from a `web3` `CallRequest`.
-    pub(crate) fn from_request(eth: Eth<T>, request: CallRequest) -> Self {
-        // NOTE(nlordell): work around issue tomusdrw/rust-web3#290; while this
-        //   bas been fixed in master, it has not been released yet
+    /// Create an estimate gas future from a `web3` `EstimateGasRequest`.
+    pub(crate) fn from_request(eth: Eth<T>, request: EstimateGasRequest) -> Self {
```

Comment:
Are we still working around the issue, or did we remove the workaround earlier without updating the comment?

Reply:
The original reason for the work around (of having our own `estimate_gas` implementation) was the because of the reason described in the comment that has now been removed.

That has since been fixed upstream. However we still need the custom `estimate_gas` implementation because of the new issue with estimating gas for contract deployment. So the custom implementation is still needed from before, but the comment is now outdated as the reason for it has changed.